### ISSUE TITLE
PIVOT statements in Databricks should allow columns without AS statements

### ISSUE THREAD

**Issue by TheCleric** [sqlfluff/sqlfluff#5949]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When parsing a Databricks SELECT statement with a PIVOT clause that has aggregate columns that have no optional AS statement, the parsing fails.

### Expected Behaviour

The parsing should succeed as long as the format matches https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-qry-select-pivot.html#syntax

### Observed Behaviour

The parsing fails with an error like the following:

`Line 36, Position 1: Found unparsable section: 'PIVOT (sum(sales)\n  FOR quarter\n  IN (1 ...'`

### How to reproduce

Here is a minimal reproducible sql example:

```sql
SELECT year, region, q1, q2, q3, q4
FROM sales
PIVOT (sum(sales)
  FOR quarter
  IN (1 AS q1, 2 AS q2, 3 AS q3, 4 AS q4));
```

### Dialect

Databricks

### Version

sqlfluff, version 3.0.7
Python 3.10.12

### Configuration

```cfg
[sqlfluff]
dialect = databricks
templater = placeholder
max_line_length = 120
exclude_rules = CP02, RF02
large_file_skip_byte_limit = 0

[sqlfluff:indentation]
tab_space_size = 2
indented_joins = True
indented_ctes = True
indented_using_on = True
indented_on_contents = True
# indented_then = True
# indented_then_contents = True
allow_implicit_indents = True
template_blocks_indent = True

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:templater:placeholder]
param_regex = \${(?P<param_name>[-\w.]+)}
ese-pipeline.catalog = ese_dev
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by TheCleric** [comment]
I'm a bit at a loss trying to fix this one myself. It seems to me as if the relevant bit is here:

https://github.com/sqlfluff/sqlfluff/blob/6c22f5ecd973ee78a42dabf7dbc5230f03a7bee4/src/sqlfluff/dialects/dialect_sparksql.py#L2064-L2072

But that shows the `AliasExpressionSegment` is already optional, so to me this shouldn't be a parsing issue. 😕

**Comment by keraion** [comment]
It looks like the difference here is going to be between Spark's and Databricks' reserved keywords. In Databricks, it moves most of Spark's reserved words to unreserved. This includes `FOR` which here follows an optional `AliasExpressionSegment` which it attempts to consume it as an alias rather than the keyword.

To fix this, an exclusion for `FOR` would need to be added to as part of the `AliasExpressionSegment`.
